samza-core/src/main/java/org/apache/samza/storage/StorageManagerUtil.java

```diff
@@ -145,10 +145,14 @@ public static boolean isOffsetFileValid(File storeDir, Set<SystemStreamPartition
     boolean hasValidOffsetFile = false;
     if (storeDir.exists()) {
       Map<SystemStreamPartition, String> offsetContents = readOffsetFile(storeDir, storeSSPs, isSideInput);
-      if (offsetContents != null && !offsetContents.isEmpty() && offsetContents.keySet().equals(storeSSPs)) {
-        hasValidOffsetFile = true;
+      if (offsetContents == null) {
+        LOG.info("Offset file is not valid for store: {}.", storeDir.toPath() + "; the file does not exist.");
```

Comment:
Don't use string-concat - Use log formatters instead:
LOG.info("Offset file does not exist. Store directory: {}", storeDir.toPath);